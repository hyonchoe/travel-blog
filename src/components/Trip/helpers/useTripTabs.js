/**
 * Hook for handling Trip component's tabs
 */

import { useState } from 'react'

export const journalKey = 'journal'
export const imageKey = 'images'
export const journalTab = { key: journalKey, tab: 'Journal' }
export const imageTab = { key: imageKey, tab: 'Photos' }

/**
 * Gets tabs to use for Trip component
 * @param {Array} tripImages Trip image data
 * @returns {Array} Tabs to use
 */
export const getTripTabList = (tripImages) => {
    let cardTabList = [journalTab]
    
    if (tripImages && tripImages.length > 0){
        cardTabList.push(imageTab)
    }

    return cardTabList
}

const useTripTabs = () => {
    const [tabState, setTabState] = useState({ key: journalKey })
    
    /**
     * Updates tab state
     * @param {string} key Tab key
     * @param {string} type Property for data
     */
    const onTabChange = (key, type) => {
        setTabState({ [type]: key })
    }

    /**
     * Gets tabs to use
     * @param {Array} tripImages Trip image data
     * @returns {Array} Tabs to use
     */
    const getTabList = (tripImages) => {
        return getTripTabList(tripImages)
    }

    return { tabState, onTabChange, getTabList }
}

export default useTripTabs