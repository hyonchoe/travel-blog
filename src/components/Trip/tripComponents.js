import React from 'react'
import { Typography } from 'antd'
import { LockOutlined } from '@ant-design/icons'
import PictureCarousel from '../PictureCarousel'
    
export const LocationSpans = (locations, cssStyle) => {
    if (!locations) return null

    return locations.map((loc) => (
        <span className={cssStyle}>{loc.fmtAddr}</span>
    ))
}

export const TravelerName = (showMyTrips, isMyTrip, userInfo) => {    
    if (showMyTrips) return ''
    
    let name = (isMyTrip) ? 'me' : (userInfo.userName) ? userInfo.userName : userInfo.userEmail
    if (!name){
        name = 'someone'
    }

    return <span>Travelled by {name}</span>
}

export const getCardTitle = (title, isPublic, cssStyle) => {
    if (isPublic){
        return <span>{title}</span>
    }

    return (
        <span>
            {title}<LockOutlined className={cssStyle} />
        </span>
    )
}

export const TripTabContent = (props) => {
    let contentComponent = null
    switch (props.tabKey){
        case 'journal':
            contentComponent = <TripDetails 
                                details={props.curTrip.details}
                                locAddr={props.locAddr}
                                travelerName={props.travelerName}
                                cssStyle={props.cssStyle} />
            break

        case 'images':
            contentComponent = <PictureCarousel images={props.curTrip.images}/>
            break

        default:
            contentComponent = <TripDetails 
                                details={props.curTrip.details}
                                locAddr={props.locAddr}
                                travelerName={props.travelerName}
                                cssStyle={props.cssStyle} />
            break
    }
    
    return contentComponent
}

export const TripDates = (curTrip) => {
    const dateFormat="dddd, MMMM Do YYYY"
    let dateStr = curTrip.startDate.format(dateFormat)
    if (curTrip.endDate){
        dateStr += " - " + curTrip.endDate.format(dateFormat)
    }    
    
    return dateStr
}

const TripDetails = (props) => {
    return (
        <Typography>
            {props.travelerName && 
            <div className={props.cssStyle.name}>{props.travelerName}</div>
            }
            {props.locAddr && 
            <div className={props.cssStyle.loc}> {props.locAddr} </div>
            }
            { 
            props.details.split('\n').map( (line) => { 
                if (line) {
                    return <p>{line}</p>
                }
                return ""
            })
            }
        </Typography>
    )
}