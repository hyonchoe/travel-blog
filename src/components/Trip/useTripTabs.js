import { useState } from 'react'

const useTripTabs = () => {
    const [tabState, setTabState] = useState({ key: 'journal' })
    
    const onTabChange = (key, type) => {
        setTabState({ [type]: key })
    }

    return { tabState, onTabChange }
}

export default useTripTabs