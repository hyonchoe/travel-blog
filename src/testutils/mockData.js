import moment from 'moment'

const mockData = () => {
    //#region Data values
    const tripId = 'dummytripid'
    const otherTripId = 'otherdummytripid'
    const userId = 'dummyuserid'
    const userName = 'dummyusername'
    const userEmail = 'dummyuseremail'
    const title = 'dummytitle'
    const isPublic = true
    const details = 'dummydetails'
    const loc1 = {
        fmtAddr: 'dummyfmtaddr1',
        latLng: [1, 1],
        city: 'dummycity1',
        state: 'dummystate1',
        country: 'dummycountry1',
    }
    const loc2 = {
        fmtAddr: 'dummyfmtaddr2',
        latLng: [2, 2],
        city: 'dummycity2',
        state: 'dummystate2',
        country: 'dummycountry2',
    }
    const locations = [loc1, loc2]
    const image1 = {
        name: 'dummyname1',
        fileUrlName: 'dummyurlname1',
        S3Url: 'dummys3url1'
    }
    const image2 = {
        name: 'dummyname2',
        fileUrlName: 'dummyurlname2',
        S3Url: 'dummys3url2'
    }
    const image3 = {
        name: 'dummyname3',
        fileUrlName: 'dummyurlname3',
        S3Url: 'dummys3url3'
    }
    const images = [image1, image2, image3]
    const startDate = moment('2020-07-01', 'YYYY-MM-DD')
    const endDate = moment('2020-07-04', 'YYYY-MM-DD')
    //#endregion

    const getTrip = () => ({
        _id: tripId,
        userId,
        userName,
        userEmail,
        title,
        startDate,
        endDate,
        public: isPublic,
        details,
        locations,
        images
    })

    const getLastTripLoaded = () => ({
        _id: tripId,
        startDate: startDate,
        endDate: endDate,
    })

    const getUserInfo = () => ({
        userName,
        userEmail,
    })

    const getAuth0UserInfo = () => ({
        name: 'dummyname',
        email: 'dummyemail',
        sub: 'dummysub'
    })

    return {
        tripId, otherTripId, title, startDate, endDate,
        images, locations, loc1,
        getTrip, getLastTripLoaded, getUserInfo,
        getAuth0UserInfo
    }
}

export default mockData