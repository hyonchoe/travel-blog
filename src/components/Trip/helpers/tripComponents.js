import React from 'react'
import { Typography, Tooltip, Popconfirm } from 'antd'
import { LockOutlined, GlobalOutlined, EditOutlined, DeleteOutlined } from '@ant-design/icons'
import { journalKey, imageKey } from './useTripTabs'
import PictureCarousel from '../../PictureCarousel'
import { onGlobeClicked, popoverConfirm, popoverCancel, showDeleteConfirm } from './cardActions'
    
export const LocationSpans = (locations, cssStyle) => {
    if (!locations) return null

    return locations.map((loc, index) => (
        <span className={cssStyle} key={index}>{loc.fmtAddr}</span>
    ))
}

export const TravelerName = (showMyTrips, isMyTrip, userInfo) => {    
    if (showMyTrips) return ''
    
    let name = (isMyTrip) ? 'me' : (userInfo.userName) ? userInfo.userName : userInfo.userEmail
    if (!name){
        name = 'someone'
    }

    return <span>Travelled by {name}</span>
}

export const CardTitle = (title, isPublic, cssStyle) => {
    if (isPublic){
        return <span>{title}</span>
    }

    return (
        <span>
            {title}<LockOutlined className={cssStyle} />
        </span>
    )
}

export const TripTabContent = (props) => {
    let contentComponent = null
    switch (props.tabKey){
        case journalKey:
            contentComponent = <TripDetails 
                                details={props.curTrip.details}
                                locAddr={props.locAddr}
                                travelerName={props.travelerName}
                                cssStyle={props.cssStyle} />
            break

        case imageKey:
            contentComponent = <PictureCarousel images={props.curTrip.images}/>
            break

        default:
            contentComponent = <TripDetails 
                                details={props.curTrip.details}
                                locAddr={props.locAddr}
                                travelerName={props.travelerName}
                                cssStyle={props.cssStyle} />
            break
    }
    
    return contentComponent
}

export const TripDates = (curTrip) => {
    const dateFormat="dddd, MMMM Do YYYY"
    let dateStr = curTrip.startDate.format(dateFormat)
    if (curTrip.endDate){
        dateStr += " - " + curTrip.endDate.format(dateFormat)
    }
    
    return dateStr
}

export const getCardActions = (curTrip, isCurUserTrip, launchMapCallback, editTripCallback, deleteCallback) => {
    const actions = []
    actions.push(
        <Tooltip title="View this trip locations on a map" key ={0}>
            <GlobalOutlined key="map" onClick={() => onGlobeClicked(curTrip, launchMapCallback) } />
        </Tooltip>
    )

    if (isCurUserTrip){
        actions.push(
            <Tooltip title="Edit this trip information" key={1}>
                <Popconfirm
                    title="Are you sure about editing this trip?"
                    onConfirm={() => popoverConfirm(curTrip, editTripCallback)}
                    onCancel={() => popoverCancel()}
                    okText="Yes"
                    cancelText="No" >
                    <EditOutlined key="edit" />
                </Popconfirm>
            </Tooltip>            
        )
        actions.push(
            <Tooltip title="Delete this trip" key={2}>
                <DeleteOutlined key="delete" onClick={() => showDeleteConfirm(curTrip._id, curTrip.title, deleteCallback)} />
            </Tooltip>
        )
    }

    return actions
}

const TripDetails = (props) => {
    return (
        <Typography>
            {props.travelerName && 
            <div className={props.cssStyle.name}>{props.travelerName}</div>
            }
            {props.locAddr && 
            <div className={props.cssStyle.loc}>{props.locAddr}</div>
            }
            { 
            props.details.split('\n').map((line, index) => { 
                if (line) {
                    return <p key={index}>{line}</p>
                }
                return ""
            })
            }
        </Typography>
    )
}