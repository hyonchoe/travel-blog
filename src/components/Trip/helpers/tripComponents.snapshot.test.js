/**
 * Snapshot tests for helper components for Trip component (tripComponents)
 * 
 * Run by running 'npm test' in command line.
 */

import React from 'react'
import renderer from 'react-test-renderer'
import { LocationSpans, TravelerName, CardTitle, TripTabContent, TripDates, getCardActions } from './tripComponents'
import { journalKey, imageKey } from './useTripTabs'
import mockData from '../../../testutils/mockData'

jest.mock('../../PictureCarousel', () => {
    const mockedComponent = () => <div className='mockedPictureCarousel' />
    return mockedComponent
})

const dummyCssClassName = 'dummyCssClassName'
const trip = mockData().getTrip()

describe('LocationSpans', () => {
    it('matches snapshot for no location', () => {
        const component = renderer.create(
            LocationSpans(null, dummyCssClassName)
        )
        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('matches snapshot for some locations', () => {
        const component = renderer.create(
            LocationSpans(mockData().locations, dummyCssClassName)
        )
        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })
})

describe('TravelerName', () => {
    it('matches snapshot for showing My Trips', () => {
        const dummyParamIsMyTrip = true
        const dummyParamUserInfo = {}
        const component = renderer.create(
            TravelerName(true, dummyParamIsMyTrip, dummyParamUserInfo)
        )
        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('matches snapshot for my trip in public trips', () => {
        const dummyParamUserInfo = {}
        const component = renderer.create(
            TravelerName(false, true, dummyParamUserInfo)
        )
        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('matches snapshot for not my trip in public trips - user name', () => {
        const userInfo = mockData().getUserInfo()
        const component = renderer.create(
            TravelerName(false, false, userInfo)
        )
        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('matches snapshot for not my trip in public trips - user email', () => {
        const userInfo = mockData().getUserInfo()
        delete userInfo.userName
        const component = renderer.create(
            TravelerName(false, false, userInfo)
        )
        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('matches snapshot for no identifiable information', () => {
        const component = renderer.create(
            TravelerName(false, false, {})
        )
        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })
})

describe('CardTitle', () => {
    const title = mockData().title

    it('matches snapshot for public trip', () => {
        const component = renderer.create(
            CardTitle(title, true, dummyCssClassName)
        )
        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('matches snapshot for private trip', () => {
        const component = renderer.create(
            CardTitle(title, false, dummyCssClassName)
        )
        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })
})

describe('TripTabContent', () => {
    const dummyLocSpans = (<span className='dummy' key={0}>dummy addr</span>)
    const dummyTravelersName = (<span>dummy traveler name</span>)
    const dummyNameCssClass = 'nameCssClass'
    const dummyLocCssClass = 'locCssClass'

    it('matches snapshot for journey tab content', () => {
        const component = renderer.create(
            <TripTabContent 
                tabKey={journalKey}
                curTrip= {trip}
                locAddr= {dummyLocSpans}
                travelerName= {dummyTravelersName}
                cssStyle= { {name: dummyNameCssClass, loc: dummyLocCssClass} } />
        )
        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })
    
   it('mathces snapshot for photo tab content (PictureCarousel component mocked)', () => {
        const component = renderer.create(
            <TripTabContent 
                tabKey={imageKey}
                curTrip= {trip}
                locAddr= {dummyLocSpans}
                travelerName= {dummyTravelersName}
                cssStyle= { {name: dummyNameCssClass, loc: dummyLocCssClass} } />
        )
        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()       
        expect(true).toBe(true)
   })
})

describe('TripDates', () => {
    it('matches snapshot for trip dates as text display', () => {
        const component = renderer.create(
            TripDates(trip, false)
        )
        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('matches snapshot for trip dates string as icon display', () => {
        const component = renderer.create(
            TripDates(trip, true)
        )
        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })
})

describe('Card actions', () => {
    const dummyParamCallback = () => {}

    it('matches snapshot for one action (map)', () => {
        const component = renderer.create(
            getCardActions(trip, false, dummyParamCallback, dummyParamCallback, dummyParamCallback)
        )
        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('matches snapshot for three actions (map, edit, delete)', () => {
        const component = renderer.create(
            getCardActions(trip, true, dummyParamCallback, dummyParamCallback, dummyParamCallback)
        )
        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })
})