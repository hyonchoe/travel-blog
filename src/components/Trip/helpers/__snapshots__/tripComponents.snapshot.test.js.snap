// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card actions matches snapshot for one action (map) 1`] = `
<span
  aria-label="global"
  className="anticon anticon-global"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  role="img"
  tabIndex={-1}
>
  <svg
    aria-hidden="true"
    className=""
    data-icon="global"
    fill="currentColor"
    focusable="false"
    height="1em"
    viewBox="64 64 896 896"
    width="1em"
  >
    <path
      d="M854.4 800.9c.2-.3.5-.6.7-.9C920.6 722.1 960 621.7 960 512s-39.4-210.1-104.8-288c-.2-.3-.5-.5-.7-.8-1.1-1.3-2.1-2.5-3.2-3.7-.4-.5-.8-.9-1.2-1.4l-4.1-4.7-.1-.1c-1.5-1.7-3.1-3.4-4.6-5.1l-.1-.1c-3.2-3.4-6.4-6.8-9.7-10.1l-.1-.1-4.8-4.8-.3-.3c-1.5-1.5-3-2.9-4.5-4.3-.5-.5-1-1-1.6-1.5-1-1-2-1.9-3-2.8-.3-.3-.7-.6-1-1C736.4 109.2 629.5 64 512 64s-224.4 45.2-304.3 119.2c-.3.3-.7.6-1 1-1 .9-2 1.9-3 2.9-.5.5-1 1-1.6 1.5-1.5 1.4-3 2.9-4.5 4.3l-.3.3-4.8 4.8-.1.1c-3.3 3.3-6.5 6.7-9.7 10.1l-.1.1c-1.6 1.7-3.1 3.4-4.6 5.1l-.1.1c-1.4 1.5-2.8 3.1-4.1 4.7-.4.5-.8.9-1.2 1.4-1.1 1.2-2.1 2.5-3.2 3.7-.2.3-.5.5-.7.8C103.4 301.9 64 402.3 64 512s39.4 210.1 104.8 288c.2.3.5.6.7.9l3.1 3.7c.4.5.8.9 1.2 1.4l4.1 4.7c0 .1.1.1.1.2 1.5 1.7 3 3.4 4.6 5l.1.1c3.2 3.4 6.4 6.8 9.6 10.1l.1.1c1.6 1.6 3.1 3.2 4.7 4.7l.3.3c3.3 3.3 6.7 6.5 10.1 9.6 80.1 74 187 119.2 304.5 119.2s224.4-45.2 304.3-119.2a300 300 0 0010-9.6l.3-.3c1.6-1.6 3.2-3.1 4.7-4.7l.1-.1c3.3-3.3 6.5-6.7 9.6-10.1l.1-.1c1.5-1.7 3.1-3.3 4.6-5 0-.1.1-.1.1-.2 1.4-1.5 2.8-3.1 4.1-4.7.4-.5.8-.9 1.2-1.4a99 99 0 003.3-3.7zm4.1-142.6c-13.8 32.6-32 62.8-54.2 90.2a444.07 444.07 0 00-81.5-55.9c11.6-46.9 18.8-98.4 20.7-152.6H887c-3 40.9-12.6 80.6-28.5 118.3zM887 484H743.5c-1.9-54.2-9.1-105.7-20.7-152.6 29.3-15.6 56.6-34.4 81.5-55.9A373.86 373.86 0 01887 484zM658.3 165.5c39.7 16.8 75.8 40 107.6 69.2a394.72 394.72 0 01-59.4 41.8c-15.7-45-35.8-84.1-59.2-115.4 3.7 1.4 7.4 2.9 11 4.4zm-90.6 700.6c-9.2 7.2-18.4 12.7-27.7 16.4V697a389.1 389.1 0 01115.7 26.2c-8.3 24.6-17.9 47.3-29 67.8-17.4 32.4-37.8 58.3-59 75.1zm59-633.1c11 20.6 20.7 43.3 29 67.8A389.1 389.1 0 01540 327V141.6c9.2 3.7 18.5 9.1 27.7 16.4 21.2 16.7 41.6 42.6 59 75zM540 640.9V540h147.5c-1.6 44.2-7.1 87.1-16.3 127.8l-.3 1.2A445.02 445.02 0 00540 640.9zm0-156.9V383.1c45.8-2.8 89.8-12.5 130.9-28.1l.3 1.2c9.2 40.7 14.7 83.5 16.3 127.8H540zm-56 56v100.9c-45.8 2.8-89.8 12.5-130.9 28.1l-.3-1.2c-9.2-40.7-14.7-83.5-16.3-127.8H484zm-147.5-56c1.6-44.2 7.1-87.1 16.3-127.8l.3-1.2c41.1 15.6 85 25.3 130.9 28.1V484H336.5zM484 697v185.4c-9.2-3.7-18.5-9.1-27.7-16.4-21.2-16.7-41.7-42.7-59.1-75.1-11-20.6-20.7-43.3-29-67.8 37.2-14.6 75.9-23.3 115.8-26.1zm0-370a389.1 389.1 0 01-115.7-26.2c8.3-24.6 17.9-47.3 29-67.8 17.4-32.4 37.8-58.4 59.1-75.1 9.2-7.2 18.4-12.7 27.7-16.4V327zM365.7 165.5c3.7-1.5 7.3-3 11-4.4-23.4 31.3-43.5 70.4-59.2 115.4-21-12-40.9-26-59.4-41.8 31.8-29.2 67.9-52.4 107.6-69.2zM165.5 365.7c13.8-32.6 32-62.8 54.2-90.2 24.9 21.5 52.2 40.3 81.5 55.9-11.6 46.9-18.8 98.4-20.7 152.6H137c3-40.9 12.6-80.6 28.5-118.3zM137 540h143.5c1.9 54.2 9.1 105.7 20.7 152.6a444.07 444.07 0 00-81.5 55.9A373.86 373.86 0 01137 540zm228.7 318.5c-39.7-16.8-75.8-40-107.6-69.2 18.5-15.8 38.4-29.7 59.4-41.8 15.7 45 35.8 84.1 59.2 115.4-3.7-1.4-7.4-2.9-11-4.4zm292.6 0c-3.7 1.5-7.3 3-11 4.4 23.4-31.3 43.5-70.4 59.2-115.4 21 12 40.9 26 59.4 41.8a373.81 373.81 0 01-107.6 69.2z"
    />
  </svg>
</span>
`;

exports[`Card actions matches snapshot for three actions (map, edit, delete) 1`] = `
Array [
  <span
    aria-label="global"
    className="anticon anticon-global"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    role="img"
    tabIndex={-1}
  >
    <svg
      aria-hidden="true"
      className=""
      data-icon="global"
      fill="currentColor"
      focusable="false"
      height="1em"
      viewBox="64 64 896 896"
      width="1em"
    >
      <path
        d="M854.4 800.9c.2-.3.5-.6.7-.9C920.6 722.1 960 621.7 960 512s-39.4-210.1-104.8-288c-.2-.3-.5-.5-.7-.8-1.1-1.3-2.1-2.5-3.2-3.7-.4-.5-.8-.9-1.2-1.4l-4.1-4.7-.1-.1c-1.5-1.7-3.1-3.4-4.6-5.1l-.1-.1c-3.2-3.4-6.4-6.8-9.7-10.1l-.1-.1-4.8-4.8-.3-.3c-1.5-1.5-3-2.9-4.5-4.3-.5-.5-1-1-1.6-1.5-1-1-2-1.9-3-2.8-.3-.3-.7-.6-1-1C736.4 109.2 629.5 64 512 64s-224.4 45.2-304.3 119.2c-.3.3-.7.6-1 1-1 .9-2 1.9-3 2.9-.5.5-1 1-1.6 1.5-1.5 1.4-3 2.9-4.5 4.3l-.3.3-4.8 4.8-.1.1c-3.3 3.3-6.5 6.7-9.7 10.1l-.1.1c-1.6 1.7-3.1 3.4-4.6 5.1l-.1.1c-1.4 1.5-2.8 3.1-4.1 4.7-.4.5-.8.9-1.2 1.4-1.1 1.2-2.1 2.5-3.2 3.7-.2.3-.5.5-.7.8C103.4 301.9 64 402.3 64 512s39.4 210.1 104.8 288c.2.3.5.6.7.9l3.1 3.7c.4.5.8.9 1.2 1.4l4.1 4.7c0 .1.1.1.1.2 1.5 1.7 3 3.4 4.6 5l.1.1c3.2 3.4 6.4 6.8 9.6 10.1l.1.1c1.6 1.6 3.1 3.2 4.7 4.7l.3.3c3.3 3.3 6.7 6.5 10.1 9.6 80.1 74 187 119.2 304.5 119.2s224.4-45.2 304.3-119.2a300 300 0 0010-9.6l.3-.3c1.6-1.6 3.2-3.1 4.7-4.7l.1-.1c3.3-3.3 6.5-6.7 9.6-10.1l.1-.1c1.5-1.7 3.1-3.3 4.6-5 0-.1.1-.1.1-.2 1.4-1.5 2.8-3.1 4.1-4.7.4-.5.8-.9 1.2-1.4a99 99 0 003.3-3.7zm4.1-142.6c-13.8 32.6-32 62.8-54.2 90.2a444.07 444.07 0 00-81.5-55.9c11.6-46.9 18.8-98.4 20.7-152.6H887c-3 40.9-12.6 80.6-28.5 118.3zM887 484H743.5c-1.9-54.2-9.1-105.7-20.7-152.6 29.3-15.6 56.6-34.4 81.5-55.9A373.86 373.86 0 01887 484zM658.3 165.5c39.7 16.8 75.8 40 107.6 69.2a394.72 394.72 0 01-59.4 41.8c-15.7-45-35.8-84.1-59.2-115.4 3.7 1.4 7.4 2.9 11 4.4zm-90.6 700.6c-9.2 7.2-18.4 12.7-27.7 16.4V697a389.1 389.1 0 01115.7 26.2c-8.3 24.6-17.9 47.3-29 67.8-17.4 32.4-37.8 58.3-59 75.1zm59-633.1c11 20.6 20.7 43.3 29 67.8A389.1 389.1 0 01540 327V141.6c9.2 3.7 18.5 9.1 27.7 16.4 21.2 16.7 41.6 42.6 59 75zM540 640.9V540h147.5c-1.6 44.2-7.1 87.1-16.3 127.8l-.3 1.2A445.02 445.02 0 00540 640.9zm0-156.9V383.1c45.8-2.8 89.8-12.5 130.9-28.1l.3 1.2c9.2 40.7 14.7 83.5 16.3 127.8H540zm-56 56v100.9c-45.8 2.8-89.8 12.5-130.9 28.1l-.3-1.2c-9.2-40.7-14.7-83.5-16.3-127.8H484zm-147.5-56c1.6-44.2 7.1-87.1 16.3-127.8l.3-1.2c41.1 15.6 85 25.3 130.9 28.1V484H336.5zM484 697v185.4c-9.2-3.7-18.5-9.1-27.7-16.4-21.2-16.7-41.7-42.7-59.1-75.1-11-20.6-20.7-43.3-29-67.8 37.2-14.6 75.9-23.3 115.8-26.1zm0-370a389.1 389.1 0 01-115.7-26.2c8.3-24.6 17.9-47.3 29-67.8 17.4-32.4 37.8-58.4 59.1-75.1 9.2-7.2 18.4-12.7 27.7-16.4V327zM365.7 165.5c3.7-1.5 7.3-3 11-4.4-23.4 31.3-43.5 70.4-59.2 115.4-21-12-40.9-26-59.4-41.8 31.8-29.2 67.9-52.4 107.6-69.2zM165.5 365.7c13.8-32.6 32-62.8 54.2-90.2 24.9 21.5 52.2 40.3 81.5 55.9-11.6 46.9-18.8 98.4-20.7 152.6H137c3-40.9 12.6-80.6 28.5-118.3zM137 540h143.5c1.9 54.2 9.1 105.7 20.7 152.6a444.07 444.07 0 00-81.5 55.9A373.86 373.86 0 01137 540zm228.7 318.5c-39.7-16.8-75.8-40-107.6-69.2 18.5-15.8 38.4-29.7 59.4-41.8 15.7 45 35.8 84.1 59.2 115.4-3.7-1.4-7.4-2.9-11-4.4zm292.6 0c-3.7 1.5-7.3 3-11 4.4 23.4-31.3 43.5-70.4 59.2-115.4 21 12 40.9 26 59.4 41.8a373.81 373.81 0 01-107.6 69.2z"
      />
    </svg>
  </span>,
  <span
    aria-label="edit"
    className="anticon anticon-edit"
    onClick={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onTouchStart={[Function]}
    role="img"
    tabIndex={-1}
  >
    <svg
      aria-hidden="true"
      className=""
      data-icon="edit"
      fill="currentColor"
      focusable="false"
      height="1em"
      viewBox="64 64 896 896"
      width="1em"
    >
      <path
        d="M257.7 752c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 000-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 009.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9zm67.4-174.4L687.8 215l73.3 73.3-362.7 362.6-88.9 15.7 15.6-89zM880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32z"
      />
    </svg>
  </span>,
  <span
    aria-label="delete"
    className="anticon anticon-delete"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    role="img"
    tabIndex={-1}
  >
    <svg
      aria-hidden="true"
      className=""
      data-icon="delete"
      fill="currentColor"
      focusable="false"
      height="1em"
      viewBox="64 64 896 896"
      width="1em"
    >
      <path
        d="M360 184h-8c4.4 0 8-3.6 8-8v8h304v-8c0 4.4 3.6 8 8 8h-8v72h72v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80h72v-72zm504 72H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zM731.3 840H292.7l-24.2-512h487l-24.2 512z"
      />
    </svg>
  </span>,
]
`;

exports[`CardTitle matches snapshot for private trip 1`] = `
<span>
  dummytitle
  <span
    aria-label="lock"
    className="anticon anticon-lock dummyCssClassName"
    role="img"
  >
    <svg
      aria-hidden="true"
      className=""
      data-icon="lock"
      fill="currentColor"
      focusable="false"
      height="1em"
      viewBox="64 64 896 896"
      width="1em"
    >
      <path
        d="M832 464h-68V240c0-70.7-57.3-128-128-128H388c-70.7 0-128 57.3-128 128v224h-68c-17.7 0-32 14.3-32 32v384c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V496c0-17.7-14.3-32-32-32zM332 240c0-30.9 25.1-56 56-56h248c30.9 0 56 25.1 56 56v224H332V240zm460 600H232V536h560v304zM484 701v53c0 4.4 3.6 8 8 8h40c4.4 0 8-3.6 8-8v-53a48.01 48.01 0 10-56 0z"
      />
    </svg>
  </span>
</span>
`;

exports[`CardTitle matches snapshot for public trip 1`] = `
<span>
  dummytitle
</span>
`;

exports[`LocationSpans matches snapshot for no location 1`] = `null`;

exports[`LocationSpans matches snapshot for some locations 1`] = `
Array [
  <span
    className="dummyCssClassName"
  >
    dummyfmtaddr1
  </span>,
  <span
    className="dummyCssClassName"
  >
    dummyfmtaddr2
  </span>,
]
`;

exports[`TravelerName matches snapshot for my trip in public trips 1`] = `
<span>
  Travelled by 
  me
</span>
`;

exports[`TravelerName matches snapshot for no identifiable information 1`] = `
<span>
  Travelled by 
  someone
</span>
`;

exports[`TravelerName matches snapshot for not my trip in public trips - user email 1`] = `
<span>
  Travelled by 
  dummyuseremail
</span>
`;

exports[`TravelerName matches snapshot for not my trip in public trips - user name 1`] = `
<span>
  Travelled by 
  dummyusername
</span>
`;

exports[`TravelerName matches snapshot for showing My Trips 1`] = `""`;

exports[`TripDates matches snapshot for trip dates as text display 1`] = `
<span>
  Wed, Jul 1st 2020 - Sat, Jul 4th 2020
</span>
`;

exports[`TripDates matches snapshot for trip dates string as icon display 1`] = `
<span
  aria-label="calendar"
  className="anticon anticon-calendar"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  role="img"
>
  <svg
    aria-hidden="true"
    className=""
    data-icon="calendar"
    fill="currentColor"
    focusable="false"
    height="1em"
    viewBox="64 64 896 896"
    width="1em"
  >
    <path
      d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
    />
  </svg>
</span>
`;

exports[`TripTabContent matches snapshot for journey tab content 1`] = `
<article
  className="ant-typography"
>
  <div
    className="nameCssClass"
  >
    <span>
      dummy traveler name
    </span>
  </div>
  <div
    className="locCssClass"
  >
    <span
      className="dummy"
    >
      dummy addr
    </span>
  </div>
  <p>
    dummydetails
  </p>
</article>
`;

exports[`TripTabContent matches snapshot for photo tab content (PictureCarousel component mocked) 1`] = `
<div
  className="mockedPictureCarousel"
/>
`;
